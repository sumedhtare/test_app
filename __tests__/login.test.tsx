import renderer from 'react-test-renderer';
import {it, describe, expect} from '@jest/globals';
import React from 'react';
import LoginScreen from '../src/screens/login';

describe('renders and test login', () => {
  it('should be present', () => {
    const tree = renderer.create(<LoginScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should log in with valid credentials', () => {
    const component = renderer.create(<LoginScreen />);
    const instance: any = component.getInstance();
    if (instance) {
      // Simulate user input
      instance.setState({username: 'admin', password: 'admin'});

      // Trigger the login function
      instance.handleLogin();

      // Assert that the user is logged in
      expect(instance.state.loggedIn).toBeTruthy();
    }
  });

  it('should not log in with invalid credentials', () => {
    const component = renderer.create(<LoginScreen />);
    const instance: any = component.getInstance();
    if (instance) {
      // Simulate user input
      instance.setState({username: 'adin', password: 'adminad'});

      // Trigger the login function
      instance.handleLogin();

      // Assert that the user is logged in
      expect(instance.state.loggedIn).toBeFalsy();
    }
  });
});
