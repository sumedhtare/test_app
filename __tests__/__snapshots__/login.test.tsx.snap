// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and test login should be present 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "gap": 15,
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      id="test-username"
      onChangeText={[Function]}
      placeholder="Username"
      style={
        {
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 40,
          "width": "80%",
        }
      }
      value=""
    />
    <TextInput
      id="test-password"
      onChangeText={[Function]}
      placeholder="Username"
      secureTextEntry={true}
      style={
        {
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 40,
          "width": "80%",
        }
      }
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="test-login"
    >
      <View
        style={
          [
            {},
            {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
